test/e2e_tests/specs/Block/block.spec.ts

```diff
@@ -17,83 +17,82 @@
-    userA = getUser();
-    userB = getUser();
+import {test, expect, withLogin, withConnectionRequest, withConnectedUser} from 'test/e2e_tests/test.fixtures';
 
-    // Step 1: Create and log in users
-    await test.step('Preconditions: Creating test users via API', async () => {
-      await api.createPersonalUser(userA);
-      addCreatedUser(userA);
+import {createGroup} from '../../utils/userActions';
 
-      await api.createPersonalUser(userB);
-      addCreatedUser(userB);
-    });
-
-    await test.step('Preconditions: Signing in User A and User B', async () => {
-      await Promise.all([loginAndSetup(userA, userAPageManager), loginAndSetup(userB, userBPageManager)]);
+/* ========== Block Utils ========== */
+/**
+ * Blocks a user from the conversation list
+ * @param pageManager PageManager of the blocking user
+ * @param userToBlock User object of the user to be blocked
+ * @param options Optional parameters, e.g. to handle additional modals
+ */
+export async function blockUserFromConversationList(pageManager: PageManager, userToBlock: User) {
+  const {pages, modals} = pageManager.webapp;
+
+  await pages.conversationList().openConversation(userToBlock.fullName);
+  await pages.conversationList().clickConversationOptions(userToBlock.fullName);
+  await pages.conversationList().clickBlockConversation();
+  await modals.blockWarning().clickBlock();
+}
+
+/**
+ * Blocks a user from their profile view (from a 1:1 conversation)
+ * @param pageManager PageManager of the blocking user
+ * @param userToBlock User object of the user to be blocked
+ */
+export async function blockUserFromProfileView(pageManager: PageManager, userToBlock: User) {
+  const {pages, modals} = pageManager.webapp;
+  await pages.conversationList().openConversation(userToBlock.fullName);
+  await pages.conversation().clickConversationInfoButton();
+  await pages.participantDetails().blockUser();
+  await modals.blockWarning().clickBlock();
+}
+
+/**
+ * Blocks a user via the participant details in a group chat
+ * Assumes that the group conversation is already open
+ * @param pageManager PageManager of the blocking user
+ * @param userToBlock User object of the user to be blocked
+ */
+export async function blockUserFromOpenGroupProfileView(pageManager: PageManager, userToBlock: User) {
+  const {pages, modals} = pageManager.webapp;
+  await pages.conversation().clickConversationTitle();
+  await pages.conversationDetails().openParticipantDetails(userToBlock.fullName);
+  await pages.participantDetails().blockUser();
+  await modals.blockWarning().clickBlock();
+}
+/* ==================================== */
+
+test.describe('User Blocking', () => {
+  test.describe('Block: User A and User B are NOT in the same team', () => {
+    let userA: User;
+    let userB: User;
+    let userC: User;
+
+    test.beforeEach(async ({createTeam}) => {
+      const teamA = await createTeam('Team A', {withMembers: 1});
```

Comment:
Since this runs before each test it would be great to parallelize the team creation:
```suggestion
      const [teamA, teamB] = await Promise.all([createTeam('Team A', {withMembers: 1}), createTeam('Team B')]);
```